src/poetry/utils/env.py

```diff
@@ -1678,6 +1688,17 @@ def get_marker_env(self) -> dict[str, Any]:
             iver = "0"
             implementation_name = ""
 
+        primary_version_regex = "\\d+\\.([a-zA-Z\\d]+(\\.[a-zA-Z\\d]+)*)"
+        local_segment_regex = "(\\+([a-zA-Z\\d]+(\\.[a-zA-Z\\d]+)*)?)?$"
+        pep_440_and_deadsnakes_regex = f"^{primary_version_regex}{local_segment_regex}$"
+        match = re.compile(pep_440_and_deadsnakes_regex).match(
+            platform.python_version()
+        )
+        if match is None:
+            raise ValueError("Python version must be PEP 440 compatible.")
+        clean_python_version = match.group()
+        clean_python_version = clean_python_version.rstrip("+")
+
```

Comment:
That feels a little over the top to me. I'd keep it simple and just strip the trailing plus. Afterwards, validation can be done via `poetry.core.constraints.version.Version.parse()` instead of the regexes.

I'm a bit torn if it's even necessary to validate the version here, since we parse it later anyway and already fail with a quite good mesage as can be seen in https://github.com/python-poetry/poetry/issues/6925#issue-1429055137. Of course, the message could be improved to say that the invalid version is the python version.